ruskeyverbs/views.py

```diff
@@ -26,19 +51,618 @@ def index(request):
+                """
+            except ObjectDoesNotExist:
+                PerformanceObj = None
+                last_interval = 1
+                easiness_factor = 2.5
+            # if the user really messed up, set next study date to 1 day out
+            if average_score*5 < 3.5:
+                last_interval = 1
+                """ otherwise set/update last_interval, easiness_factor in accordance with
+                the supermemo2 algorithm"""
+            else:
+                if last_interval == 1:
+                    last_interval = 2
+                elif last_interval == 2:
+                    last_interval = 6
+                else:
+                    easiness_factor += decimal.Decimal(0.1-(5-(average_score*5))*(0.08+(5-(average_score*5))*0.02))
+                    if easiness_factor < 1.3:
+                        easiness_factor = 1.3
+                    elif easiness_factor > 5:
+                        easiness_factor = 5
+                    last_interval *= easiness_factor
+                    last_interval = int(last_interval)
+            # if the user has a PerformancePerExample for this example, update it
+            if PerformanceObj:
+                PerformanceObj.easiness_factor = easiness_factor
+                PerformanceObj.last_interval = last_interval
+                PerformanceObj.date_last_studied = datetime.date.today()
+                PerformanceObj.due_date = (datetime.date.today()
+                                           + datetime.timedelta(
+                                               days=last_interval))
+                PerformanceObj.save()
+            # otherwise create one
+            else:
+                PerformanceObj = PerformancePerExample(example=example_inst,
+                                                       user=request.user,
+                                                       easiness_factor=easiness_factor,
+                                                       last_interval=last_interval,
+                                                       date_last_studied=datetime.date.today(),
+                                                       due_date=(datetime.date.today()
+                                                                 + datetime.timedelta(days=last_interval)))
+                PerformanceObj.save()
+            # after update, determine which example is due next
+            pk = Example.objects.annotate(
+                due=Min('performanceperexample__due_date',
+                        filter=Q(
+                            performanceperexample__user_id=request.user.pk)
+                        )
+                ).filter(verb=verb).order_by('due')[0].pk
+            """
+            If the user has previously studied the next example, send them right
+            into the quiz; otherwise, send them to the study page before the
+            quiz
+            """
+            try:
```

Comment:
Unless you've got a good reason all of these try/excepts to test for existence should be replaced with .exists() queries